public/app/features/logs/utils.ts

```diff
@@ -71,6 +103,25 @@ export function getLogLevelFromKey(key: string | number): LogLevel {
   if (level) {
     return level;
   }
+  // Support z/OS single-letter severity keys like 'I','W','E','S'
```

Comment:
4 lines above, you can see that levels are taken from a LogLevel object that contains the expected levels, so the swtich below is not needed.